automation/validate-pvc-name-stability.py

```diff
@@ -0,0 +1,105 @@
+#! python
+
+import logging
+import os
+import os.path
+import yaml
+import subprocess
+import sys
+
+from kubernetes import client, config
+from openshift.dynamic import DynamicClient
+
+
+def validatePVCNames(path, liveTemplates):
+    templates = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
+    errors = []
+    for templateFilename in templates:
+        with open(os.path.join(path, templateFilename), 'r') as stream:
+            try:
+                template = yaml.safe_load(stream)
+
+                if template == None:
+                    logging.info("Empty template file: %s", template)
+                    continue
+
+                templateName = template["metadata"]["name"]
+                logging.info("Checking PVC name stability for: {}".format(templateName))
+
+                pvcName = getPVCNameFrom(template)
+                pvcNamespace = getPVCNamespaceFrom(template)
+
+                matchingLiveTemplate = liveTemplates.get(templateName, None)
+                if matchingLiveTemplate:
+                    liveTemplatePVCName = getPVCNameFrom(matchingLiveTemplate)
+                    liveTemplatePVCNamespace = getPVCNamespaceFrom(matchingLiveTemplate)
+
+                    if pvcName != liveTemplatePVCName:
+                        errors.append(pvcNameErrorMessage(pvcName, templateName))
+                    if pvcNamespace != liveTemplatePVCNamespace:
+                        errors.append(pvcNamespaceErrorMessage(pvcNamespace, templateName))
+                else:
+                    logging.info("Missing liveTemplate for {}".format(templateName))
+
+            except yaml.YAMLError as exc:
+                raise exc
+            except Exception as exc:
+                logging.error(exc)
+
+    if len(errors) > 0:
+        error_message = "\n".join(errors)
+        logging.info(FAILED_INFO_MESSAGE)
+        raise Exception(error_message)
+
+
+def getParamFrom(template, paramName):
+    for param in template["parameters"]:
+        if param["name"] == paramName:
+            return param["value"]
+
+
+def getPVCNameFrom(template):
+    return getParamFrom(template, "SRC_PVC_NAME")
+
+
+def getPVCNamespaceFrom(template):
+    return getParamFrom(template, "SRC_PVC_NAMESPACE")
+
+
+def fetchLiveTemplates():
+    k8s_client = config.new_client_from_config()
+    dyn_client = DynamicClient(k8s_client)
+
+    template_v1 = dyn_client.resources.get(api_version='template.openshift.io/v1', kind='Template')
+    templates = template_v1.get()
+
+    if not templates:
+        return {}
+
+    liveTemplates = {}
+    for template in templates.items:
+        liveTemplates[template["metadata"]["name"]] = template
```

Comment:
There may be a template with the same name in different namespaces, in that case only one of them will be in the dictionary. To fix it, you could use a tuple as the key:
`(namespace, name)`.

Reply:
Right now it only fetches templates from one namespace.
Should it take them from different namespaces? As I thought all our common-templates land in one.
If yes, I agree with you.

Reply:
In the context of the tests, we only deploy the templates in a single namespace, so I don't mind if it stays this way, please decide what you intend to do so we can approve this

Reply:
I would keep it like it is then.